test/e2e/broker_default_test.go

```diff
@@ -43,135 +42,229 @@ const (
 	eventType2        = "type2"
 	eventSource1      = "source1"
 	eventSource2      = "source2"
+	nilString         = "nilstring"
+	extensionName     = `myextname`
+	extensionValue    = `myextval`
 )
 
-// eventTypeAndSource specifies the type and source of an Event.
-type eventTypeAndSource struct {
-	Type   string
-	Source string
+type eventMeta struct {
+	Type           string
+	Source         string
+	ExtensionName  string
+	ExtensionValue string
 }
 
 // Helper struct to tie the type and sources of the events we expect to receive
 // in subscribers with the selectors we use when creating their pods.
 type eventReceiver struct {
-	typeAndSource eventTypeAndSource
-	selector      map[string]string
+	meta     eventMeta
+	selector map[string]string
 }
 
 // This test annotates the testing namespace so that a default broker is created.
 // It then binds many triggers with different filtering patterns to that default broker,
 // and sends different events to the broker's address. Finally, it verifies that only
 // the appropriate events are routed to the subscribers.
 func TestDefaultBrokerWithManyTriggers(t *testing.T) {
-	client := setup(t, true)
-	defer tearDown(client)
-
-	// Label namespace so that it creates the default broker.
-	if err := client.LabelNamespace(map[string]string{"knative-eventing-injection": "enabled"}); err != nil {
-		t.Fatalf("Error annotating namespace: %v", err)
+	tests := []struct {
+		name            string
+		eventsToReceive []eventReceiver // These are the event types and sources that triggers will listen to, as well as the selectors
```

Comment:
`These are the event context attributes and extension attributes that triggers will listen to, to set in the subscriber and services pod` ??